### PR TITLE
Prepare QUBOLib 0.1.1 release

### PR DESCRIPTION
**Pull request by bernalde**
Summary
- Bump QUBOLib to 0.1.1 so a new release can carry the existing `QUBOTools = "0.10, 0.12"` compat.
- Add project metadata coverage for the package version and `QUBOTools v0.12.1` compatibility.
- Update build release-tag tests to derive expected data tags from the current package version.

Tests
- `julia --project=. -e 'import Pkg; Pkg.test()'` passed: 99/99 tests, with `QUBOTools v0.12.1` selected in the test environment.
- `julia --project=docs ./docs/make.jl --skip-deploy` passed. Documenter reported existing noncanonical docstring warnings.
- `make test-build` passed: 99/99 tests.
- `git diff --check` passed.
- `julia --startup-file=no -e '...'` passed: resolved `QUBOLib 0.1.1` with `QUBOTools 0.12.1` in a temporary environment.

Notes
- No documented lint, type-check, or format command was found beyond the Makefile and CI Julia checks above.
- General registry registration still needs to happen after the release commit is merged and tagged.

Refs #10

### CODE REVIEW

**Review by bernalde — COMMENTED** [review]
Blocking issues: None.

Nonblocking issues: None.

Questions: None.

Tests run and outcomes:
- `git diff --shortstat origin/main...refs/remotes/origin/pr/28/head`: matched GitHub totals, 4 files changed, 29 insertions, 5 deletions.
- `julia --project=. -e 'import Pkg; Pkg.test()'`: passed, 99/99, with `QUBOTools v0.12.1` selected.
- `julia --project=docs ./docs/make.jl --skip-deploy`: passed; existing noncanonical docstring warnings only.
- `make test-build`: passed, 99/99.
- `git diff --check origin/main...refs/remotes/origin/pr/28/head`: passed.
- Temporary resolver proof: passed, resolved `QUBOLib 0.1.1` with `QUBOTools 0.12.1`.
- `gh pr checks 28 --repo JuliaQUBO/QUBOLib.jl`: CI Julia matrix and docs build passed; deploy skipped for PR.

Merge-readiness: I found no issues that would block merge. Because I am the PR author, GitHub will not accept an approval from me, so this is posted as a comment and formal approval must come from another maintainer.

File: test/project.jl [hunk 1]
```diff
@@ -0,0 +1,18 @@
+function _compat_allows(compat::AbstractString, version::VersionNumber)
+    specs = split(compat, ',')
+
+    return any(specs) do spec
+        version in QUBOLib.Pkg.Types.VersionSpec(strip(spec))
+    end
```
Comment (bernalde) [inline test/project.jl]: Nonblocking. `_compat_allows` reimplements what `Pkg.Types.semver_spec` already does — that function takes the full comma-separated compat string and returns a `VersionSpec` whose `in` semantics match Pkg's resolver:

```julia
function _compat_allows(compat::AbstractString, version::VersionNumber)
    return version in QUBOLib.Pkg.Types.semver_spec(compat)
end
```

That drops the manual `split`/`strip` and stays closer to the resolver behaviour the test is trying to assert. Both forms accept the current `"0.10, 0.12"` string and agree on the boundary cases I exercised (`0.10.0`, `0.11.0`, `0.12.0`, `0.12.1`, `0.13.0`, `0.9.5`), so this is a readability/intent change only.

**Review by bernalde — COMMENTED** [review]
Review of release-prep PR.

## Summary

1. Blocking issues: none.
2. Nonblocking issues:
   - `_compat_allows` in `test/project.jl` reinvents `Pkg.Types.semver_spec`, which already understands comma-separated compat strings natively. See inline comment.
3. Questions:
   - Why a patch bump (`0.1.1`) rather than a minor (`0.2.0`)? The `QUBOTools` compat already on `main` was `0.10, 0.12`, so this PR ships no API change of its own — patch is defensible — but the release will be the first to actually expose users to the `QUBOTools 0.12` range (a minor jump from `0.10` upstream). A minor bump would also be defensible if a maintainer wants to signal the upstream-dep expansion. Not a blocker either way; just flagging for the release notes / registry PR.
4. Tests run and outcomes (clean checkout of `pr-28-head` at `68cd97d`):
   - `git diff --shortstat origin/main...pr-28-head`: 4 files changed, 29 insertions(+), 5 deletions(-). Matches the PR-reported totals.
   - `julia --project=. -e 'using Pkg; Pkg.test()'`: 99/99 pass; `QUBOTools v0.12.1` selected; the new `Project metadata` testset reports 2/2.
   - `make test-build`: 99/99 pass.
   - `julia --project=docs ./docs/make.jl --skip-deploy`: builds with only pre-existing noncanonical-docstring warnings; reports `Automatic version="0.1.1" for inventory from ../Project.toml`.
   - Sanity-checked `_compat_allows("0.10, 0.12", v)` against `Pkg.Types.semver_spec("0.10, 0.12")`; the two agree on the boundary cases (`0.10.0` accepted, `0.11.0` rejected, `0.12.0` accepted, `0.12.1` accepted, `0.13.0` rejected, `0.9.5` rejected).
   - `gh pr checks 28`: Julia 1 / 1.10 × {ubuntu, windows} matrices and the `build` job all pass; `deploy` is skipped for PRs as expected.
5. Merge-readiness: no blocking issues; the patch is small, focused, and exercised by both the main and build test suites.

Because I am the PR author, GitHub will not accept an `APPROVE` or `REQUEST_CHANGES` from me, so this is posted as a `COMMENT` and the formal approval must come from another maintainer.

File: test/project.jl (hunk 1 above)
Comment (bernalde) [reply test/project.jl]: Addressed in https://github.com/JuliaQUBO/QUBOLib.jl/commit/01bb02db093ca9c7afa8e82f82248922cb4c3b22. `test/project.jl` now delegates compat parsing to `QUBOLib.Pkg.Types.semver_spec(compat)`, and the targeted metadata test plus the full package/build/docs checks pass.
